lib/haven/cw_haven.dart

```diff
@@ -307,6 +307,23 @@ class CWHaven extends Haven {
     return havenTransactionInfo.accountIndex;
   }
 
+  @override
+  Future<void> backupHavenSeeds(Box<HavenSeedStore> havenSeedStore) async {
+    final walletInfoSource = await CakeHive.openBox<WalletInfo>(WalletInfo.boxName);
+    final wallets = walletInfoSource.values
+      .where((element) => element.type == WalletType.haven);
+    for (var w in wallets) {
+      final walletService = HavenWalletService(walletInfoSource);
+      final flutterSecureStorage = secureStorageShared;
+      final keyService = KeyService(flutterSecureStorage);
+      final password = await keyService.getWalletPassword(walletName: w.name);
+      final wallet = await walletService.openWallet(w.name, password);
+      await havenSeedStore.add(HavenSeedStore(id: wallet.id, seed: wallet.seed));
+      wallet.close();
+    }
+    await havenSeedStore.flush();
+  }
```

Comment:
We can - but we still would need to await it, so it would still prevent the app from starting while it's doing that.

Moving it to separate thread without awaiting is also possible, but we are risking not backing up the seed, and marking the migration as completed at the same time.